### ISSUE TITLE
Examples of tf$learn not working

### ISSUE THREAD

**Issue by dfalbel** [rstudio/tensorflow#64]
I'm trying to run the example in this [file](https://github.com/rstudio/tensorflow/blob/master/inst/examples/tflearn/iris_dnn.R) and it's returning a lot of warnings and errors see log below.

My session info:

```
> sessionInfo()
R version 3.2.3 (2015-12-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04 LTS

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_0.5.0      purrr_0.2.2      readr_0.2.2      tidyr_0.6.0      tibble_1.1       ggplot2_2.1.0   
[7] tidyverse_1.0.0  tensorflow_0.3.0 Matrix_1.2-3    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.8      lattice_0.20-33  assertthat_0.1   R6_2.1.2         grid_3.2.3       plyr_1.8.3      
 [7] DBI_0.4-1        gtable_0.2.0     magrittr_1.5     scales_0.4.0     tools_3.2.3      munsell_0.4.3   
[13] colorspace_1.2-6
> tf$VERSION
[1] "0.12.1"
```

Log:

```
> library(tensorflow)
> 
> # Here we just use all data as both training and testing (cheating) but
> # in practice you should do your own sampling, such as stratified sampling
> train_inds <- 1:150
> test_inds <- 1:150
> 
> temp_model_dir <- tempfile()
> dir.create(temp_model_dir)
> 
> datasets <- tf$contrib$learn$datasets$load_dataset("iris")
> 
> # Infer the real-valued feature columns
> feature_columns <- tf$contrib$learn$infer_real_valued_columns_from_input(datasets$data)
WARNING:tensorflow:float64 is not supported by many models, consider casting to float32.
> 
> # Initialize a DNN classifier with hidden units 10, 15, 10 in each layer
> classifier <- tf$contrib$learn$DNNClassifier(
+   feature_columns = feature_columns,
+   hidden_units = c(10L, 15L, 10L),
+   n_classes = 3L,
+   model_dir = temp_model_dir)
> 
> # Train a DNN Classifier
> classifier$fit(datasets$data[train_inds, ], datasets$target[train_inds], steps = 100)
WARNING:tensorflow:From /home/servicedesk/.local/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/dnn.py:315 in fit.: calling BaseEstimator.fit (from tensorflow.contrib.learn.python.learn.estimators.estimator) with x is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:From /home/servicedesk/.local/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/dnn.py:315 in fit.: calling BaseEstimator.fit (from tensorflow.contrib.learn.python.learn.estimators.estimator) with y is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:From /home/servicedesk/.local/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/dnn.py:315 in fit.: calling BaseEstimator.fit (from tensorflow.contrib.learn.python.learn.estimators.estimator) with batch_size is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:float64 is not supported by many models, consider casting to float32.
WARNING:tensorflow:*******************************************************
WARNING:tensorflow:TensorFlow's V1 checkpoint format has been deprecated.
WARNING:tensorflow:Consider switching to the more efficient V2 format:
WARNING:tensorflow:   `tf.train.Saver(write_version=tf.train.SaverDef.V2)`
WARNING:tensorflow:now on by default.
WARNING:tensorflow:*******************************************************
WARNING:tensorflow:*******************************************************
WARNING:tensorflow:TensorFlow's V1 checkpoint format has been deprecated.
WARNING:tensorflow:Consider switching to the more efficient V2 format:
WARNING:tensorflow:   `tf.train.Saver(write_version=tf.train.SaverDef.V2)`
WARNING:tensorflow:now on by default.
WARNING:tensorflow:*******************************************************
DNNClassifier
> 
> # Generate predictiosn on new data
> predictions <- classifier$predict(datasets$data[test_inds, ])
WARNING:tensorflow:From /home/servicedesk/.local/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/dnn.py:348 in predict.: calling BaseEstimator.predict (from tensorflow.contrib.learn.python.learn.estimators.estimator) with x is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:From /home/servicedesk/.local/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/dnn.py:348 in predict.: calling BaseEstimator.predict (from tensorflow.contrib.learn.python.learn.estimators.estimator) with batch_size is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:From /home/servicedesk/.local/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/estimators/dnn.py:348 in predict.: calling BaseEstimator.predict (from tensorflow.contrib.learn.python.learn.estimators.estimator) with as_iterable is deprecated and will be removed after 2016-12-01.
Instructions for updating:
Estimator is decoupled from Scikit Learn interface by moving into
separate class SKCompat. Arguments x, y and batch_size are only
available in the SKCompat class, Estimator will only accept input_fn.
Example conversion:
  est = Estimator(...) -> est = SKCompat(Estimator(...))
WARNING:tensorflow:float64 is not supported by many models, consider casting to float32.
> accuracy <- sum(predictions == datasets$target[test_inds]) / length(predictions)
Error in predictions == datasets$target[test_inds] : 
  comparison (1) is possible only for atomic and list types
> print(paste0("The accuracy is ", accuracy))
Error in print(paste0("The accuracy is ", accuracy)) : 
  error in evaluating the argument 'x' in selecting a method for function 'print': Error in paste0("The accuracy is ", accuracy) : 
  object 'accuracy' not found
```

**Comment by jjallaire** [comment]
@terrytangyuan I didn't get this error in TF 0.10 but now that I have updated to 0.12.1 I also see the same error. Does this example need to be updated to be compatible with 0.12.1?

**Comment by terrytangyuan** [comment]
Fixed in #65. Yeah the default has changed to return iterables in Python API.
